ghcide/src/Development/IDE/Core/Shake.hs

```diff
@@ -1125,6 +1128,14 @@ defineEarlyCutOffNoFile recorder f = defineEarlyCutoff recorder $ RuleNoDiagnost
     if file == emptyFilePath then do (hash, res) <- f k; return (Just hash, Just res) else
         fail $ "Rule " ++ show k ++ " should always be called with the empty string for a file"
 
+data SourceFileOrigin = FromProject | FromDependency
+
+getSourceFileOrigin :: NormalizedFilePath -> SourceFileOrigin
+getSourceFileOrigin f =
+    case isInfixOf ".hls/dependencies" (show f) of
```

Comment:
`isInfixOf` and `show` are the wrong functions to use here.

You probably want to inspect the path using functions from `System.FilePath` and use `fromNormalizedFilePath` to convert `NormalizedFilePath` to `FilePath`.